sdk/core/core-rest-pipeline/src/util/sanitizer.ts

```diff
@@ -119,6 +119,15 @@ export class Sanitizer {
           // When using sendOperationRequest, the request carries a massive
           // field with the autorest spec. No need to log it.
           return undefined;
+        } else if (
+          key === "_httpMessage" ||
+          key === "_readableState" ||
+          key === "_writableState" ||
+          key === "_parent" ||
+          key === "_tlsOptions"
+        ) {
+          // some proxy agent internal properties that we don't need to log.
```

Comment:
do we have to worry about them adding more properties? Should we avoid logging the entire agent itself somehow?

Reply:
Hm, yeah, it's too bad the serialize function doesn't give you the full path and only the local property name.

Reply:
More new properties wouldn't be worse than today's behavior even though we don't add them into the ignore list.  I had thought about ignoring all underscored properties but was afraid that it is too restricting.  I could remove this part of change too and only include checks for empty string

Reply:
Updated PR to only fix empty url issue.